### PR TITLE
Addition of the Hamilton 2022 dataset

### PR DESCRIPTION
**Pull request by JTeijema**
# Add a new dataset
This is a request for the addition of a new dataset. Below is metadata regarding the dataset.

## Dataset information
This is a multilingual dataset spanning English, German, Italian, and French research literature. It accompanies a systematic review on the topic of using music as part of teaching language to children aged 2 through 18.

## Dataset Reference
The dataset is from a paper currently in preprint, titled `"The effectiveness of using songs for teaching second or foreign languages to primary and secondary school learners: a systematic review."`

### Authors
      "Hamilton, C.",
      "Schulz, J.",
      "Chalmers, H.",
      "Murphy V."

### Reference URL
TBD

### Doi
TBD

### Dataset URL
https://osf.io/3u982/

### Dataset Name
Hamilton_2022

### License
CC-By 4.0

### CODE REVIEW

File: datasets/Hamilton_2022/etl_doi.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+# identify the used columns
+usecols=["title", "abstract", "issn", "notes"]
+
+# load the data from osf
+df = pd.read_csv(file_location, usecols=usecols)
+
+# notes to label
+regex = r'(?<=RAYYAN-INCLUSION: {)[a-zA-Z"=>, .]+'
+df["full_labels"] = [re.search(regex, note).group() for note in df.notes]
```
Comment (J535D165) [inline datasets/Hamilton_2022/etl_doi.py:15]: `pandas.str.extract` (see PRs by @gimoAI )

File: datasets/Hamilton_2022/etl_doi.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+df["label"] = [1 if "Included" in note else 0 for note in df.full_labels]
+
+# save results to file
+df[["title", "abstract", "label"]].to_csv("Hamilton_2022.csv", index=False)
```
Comment (J535D165) [inline datasets/Hamilton_2022/etl_doi.py:19]: This is the current format. Can you also add the future format? See PRs of @gimoAI.

**Review by J535D165 — COMMENTED** [review]
Thanks, see comments.

File: datasets/Hamilton_2022/Hamilton_2022.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+    "dataset_id": "Hamilton_2022",
+    "url": "",
```
Comment (J535D165) [inline datasets/Hamilton_2022/Hamilton_2022.json:3]: Educated guess?

Comment (JTeijema) [reply datasets/Hamilton_2022/Hamilton_2022.json:3]: I have sent out an email requesting a link. I'll update with new info

Comment (JTeijema) [reply datasets/Hamilton_2022/Hamilton_2022.json:3]: So, what happens when a dataset doesn't have any other page than the OSF page? @J535D165

**Comment by JTeijema** [comment]
This dataset consists mostly out of records that link to book chapters, and it will be difficult getting it in the standard DOI + label format we are aiming for in next versions. Let's close this PR for now, and keep it as an option for a later date.

**Pull request closed without merging**